### PR TITLE
Migration guidance

### PR DESCRIPTION
**Pull request by pauldougan**
Initial version of some migration guidance for tenants migrating to another platform

## how to review
- clone the branch run locally using the instructions in the README
- check the copy for typos and syntax
- ensure all links work correctly 

## who can review

- anyone but @pauldougan 
- @wryobservations

### CODE REVIEW

File: pages/content/migration-guidance.mdx [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+
+The Government Digital Service (GDS) has provided GOV.UK PaaS since 2015, supporting and providing a public cloud platform for departments using a shared hosting and responsibility model. 
+
+Following an extensive analysis period, GDS has concluded that, while the platform has been successful in its aims, the underlying technology would now require investment before it could meet its goals in the long term. 
+
+Faced with this need for re-investment GDS has decided to [decommission the platform](https://gds.blog.gov.uk/2022/07/12/why-weve-decided-to-decommission-gov-uk-paas-platform-as-a-service/), in order to focus its budget and energy on other GDS products for common use by Government.
+
+GDS will not be providing a replacement hosting service.
+
+Tenants should look to their own department, or parent department, and use existing patterns and support available for public cloud hosting services.
```
Comment (LeePorte) [inline pages/content/migration-guidance.mdx:21]: Where are these existing patterns? They will be asked for and they need to be public.

File: pages/content/migration-guidance.mdx [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+
+GDS will not be providing a replacement hosting service.
+
+Tenants should look to their own department, or parent department, and use existing patterns and support available for public cloud hosting services. 
+
+In deciding where to host their service next, tenants should consider the following principles from the [Central Digital & Data Office (CDDO)](https://www.gov.uk/government/organisations/central-digital-and-data-office/about#responsibilities):
+
+- Tenants should continue to use public cloud services, or internal platforms based on public cloud offerings. These continue to be the platforms of choice for government services.
+- The [Central Digital & Data Office (CDDO)](https://cddo.blog.gov.uk/) has secured Director General level agreement to help share services. Larger departments are prepared to share knowledge, patterns and platforms with the rest of government. If tenants have no existing cloud engineering capability, you could approach the CDDO, via the GOV.UK PaaS team [support form](https://admin.london.cloud.service.gov.uk/support/contact-us), to benefit from the sharing agreements.
```
Comment (LeePorte) [inline pages/content/migration-guidance.mdx:26]: The PaaS team are not going to be a mail forwarding service for an unknown service offering to an unknown entity.

**Review by LeePorte — DISMISSED** [review]

File: pages/content/migration-guidance.mdx (hunk 1 above)
Comment (wryobservations) [reply pages/content/migration-guidance.mdx:21]: Paul asked Phil a few times for something to link to here, suggest we remove the second half of the sentence while this is still unclear.

File: pages/content/migration-guidance.mdx (hunk 2 above)
Comment (wryobservations) [reply pages/content/migration-guidance.mdx:26]: there was a lot of debate about how to handle this - Phil's team suggested an individual's email address we can use while they set up a shared inbox.

File: pages/content/migration-guidance.mdx [hunk 3]
```diff
@@ -0,0 +1,504 @@
+---
+title: "Migrating from GOV.UK PaaS"
+---
+
+# Migrating from GOV.UK PaaS
+
+## Last updated
+
+- 1st November 2022 - first published
```
Comment (wryobservations) [inline pages/content/migration-guidance.mdx:9]: 1 November - house style

File: pages/content/migration-guidance.mdx [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+
+All platform and tenant data will be permanently deleted by 28 February 2024. 
+
+If you miss the deadline, and you need to migrate your applications, contact us via [the normal support routes](https://admin.london.cloud.service.gov.uk/support). If you contact us before we have deleted all tenant data, we will be able to provide this data. We will not be able to provide access to the applications or their configuration on GOV.UK PaaS.
+
+### Timeline for decommissioning GOV.UK PaaS
+
+These are some key dates in the lifetime of the platform until decommission.
+
+<table class="govuk-table">
```
Comment (kr8n3r) [inline pages/content/migration-guidance.mdx:54]: `class` should be `className` but that can be fixed at the end

Comment (pauldougan) [reply pages/content/migration-guidance.mdx:54]: I took this straight out of https://design-system.service.gov.uk/components/table/ but can fix that

File: pages/content/migration-guidance.mdx (hunk 3 above)
Comment (pauldougan) [reply pages/content/migration-guidance.mdx:9]: I will normalise all date references as discussed and make sure quarterly references are in the format QQ YYYY

File: pages/content/migration-guidance.mdx [hunk 5]
```diff
@@ -0,0 +1,504 @@
...
+
+By 23 December 2023 all services hosted on GOV.UK PaaS will need to have migrated to an alternate hosting platform.
+
+Any services that are not hosted on another platform by that date will go offline. 
+
+All platform and tenant data will be permanently deleted by 28 February 2024. 
+
+If you miss the deadline, and you need to migrate your applications, contact us via [the normal support routes](https://admin.london.cloud.service.gov.uk/support). If you contact us before we have deleted all tenant data, we will be able to provide this data. We will not be able to provide access to the applications or their configuration on GOV.UK PaaS.
+
+### Timeline for decommissioning GOV.UK PaaS
```
Comment (pauldougan) [inline pages/content/migration-guidance.mdx:50]: @wryobservations how about changing the title of this section to "Key Dates" I think this better describes what is is

File: pages/content/migration-guidance.mdx [hunk 6]
```diff
@@ -0,0 +1,504 @@
...
+
+# Migrating from GOV.UK PaaS
+
+## Last updated
+
+- 1st November 2022 - first published
+
+## Government cloud hosting strategy
+
+The Government Digital Service (GDS) has provided GOV.UK PaaS since 2015, supporting and providing a public cloud platform for departments using a shared hosting and responsibility model.
```
Comment (pauldougan) [inline pages/content/migration-guidance.mdx:13]: @LeePorte @wryobservations the digital marketplace came onboard in july 2016 so I'm inclined to change this to 2016

Comment (wryobservations) [reply pages/content/migration-guidance.mdx:13]: I don't have a strong view, altho we've always used the alpha as a measure of time in other write ups on this.

File: pages/content/migration-guidance.mdx [hunk 7]
```diff
@@ -0,0 +1,504 @@
...
+- Consider running some technical spikes to evaluate alternatives and to inform your decision
+
+## Ongoing support for your service
+
+Once your service is migrated to a new hosting platform you will need to look after it on an ongoing basis for a long time, potentially many years.
+
+- How is your team going to support your service over the longer term? Will there be a long lived team?
+- Do you need and can you afford to offer out of hours support, including cyber security?
+- Consider how you are going to support debugging of your applications (GOV.UK PaaS permits direct SSH access to containers but other hosting providers may not)
+- Consider how you are going to secure an engineer’s access to backing services to allow you to support your service.
```
Comment (wryobservations) [inline pages/content/migration-guidance.mdx:388]: stray full stop

File: pages/content/migration-guidance.mdx [hunk 8]
```diff
@@ -0,0 +1,504 @@
...
+<td class="govuk-table__cell">e.g. the DFE technical guidance.</td>
+</tr>
+</tbody>
+</table>
+
+
+### Approach
+
+- Keep it simple
+- Find the largest building blocks you can and consider solutions that take care of the underlying infrastructure, allowing you to focus on the specifics of your service.
```
Comment (wryobservations) [inline pages/content/migration-guidance.mdx:377]: stray full stop

File: pages/content/migration-guidance.mdx [hunk 9]
```diff
@@ -0,0 +1,504 @@
...
+
+- AWS Copilot (to provision ECS)
+- AWS ECS on Fargate
+- AWS Lambda
+- Google App Engine 
+- Google Cloud Run
+- Heroku 
+- Microsoft Azure App Service
+
+These are just some examples rather than endorsements.
```
Comment (wryobservations) [inline pages/content/migration-guidance.mdx:299]: Add: Make sure things work for your use-case (or some such words). Feedback from BEIS was that people will run with what look like instructions or recommendations.

File: pages/content/migration-guidance.mdx [hunk 10]
```diff
@@ -0,0 +1,504 @@
...
+- Can you raise a business case for the year 2023/2024?
+- What is the timeframe for your migration?
+- Are there any events or features that may constraint your migration plan? For example if your service has a - higher number of users in April, you may decide to migrate a month after when it will have less of an impact.
+- Can you identify a sequence to migrate the services if you have more than one?
+- You’ll need to explain to stakeholders that any data migration will incur some downtime. You can avoid this if absolutely necessary (see the Data migration section below).
+
+### Review your skill sets
+
+- Do you have the skills to migrate your services available in your department? You will need: 
+  - people with the skills to work with the infrastructure (defined in the DDAT framework as an [infrastructure devops engineer](https://www.gov.uk/guidance/development-operations-devops-engineer) role and also referred to as [Site Reliability Engineering](https://en.wikipedia.org/wiki/Site_reliability_engineering) in the wider technology marketplace), to enable the migration, change your pipelines§
```
Comment (wryobservations) [inline pages/content/migration-guidance.mdx:266]: stray special character

File: pages/content/migration-guidance.mdx (hunk 1 above)
Comment (pauldougan) [reply pages/content/migration-guidance.mdx:21]: ok I have trimmed the sentence

File: pages/content/migration-guidance.mdx (hunk 2 above)
Comment (pauldougan) [reply pages/content/migration-guidance.mdx:26]: ok have replaced the link to the team form with a mailto: link

File: pages/content/migration-guidance.mdx [hunk 11]
```diff
@@ -0,0 +1,497 @@
...
+<tr className="govuk-table__row">
+<td className="govuk-table__cell">End of the financial year</td>
+<td className="govuk-table__cell">31 March 2023</td>
+<td className="govuk-table__cell">After this date you may have additional funds available from your
+department to help the migration.</td>
+</tr>
+<tr className="govuk-table__row">
+<td className="govuk-table__cell">Automatic upgrade of PostgreSQL 10 databases to PostgreSQL 14</td>
+<td className="govuk-table__cell">17 April 2023 - 18 July 2022</td>
+<td className="govuk-table__cell">From 17 April and 18 July any PostgreSQL 10 plans will be
```
Comment (AP-Hunt) [inline pages/content/migration-guidance.mdx:96]: Suggest

> Between 17 April and 18 July, AWS will forcibly upgrade any PostgreSQL 10 databases to PostgreSQL 14 during a maintenance window. GOV.UK PaaS has no plans to support PostgreSQL 14. Any databases forcibly upgraded by AWS will still be functional, but you will not be able to manage them through CloudFoundry, other than to delete them or connect via Conduit. We recommend upgrading your PostgreSQL 10 databases to at least PostgreSQL 11 before 17 April if you expect to be with us by that point.

**Review by AP-Hunt — CHANGES_REQUESTED** [review]

File: pages/content/migration-guidance.mdx (hunk 11 above)
Comment (pauldougan) [reply pages/content/migration-guidance.mdx:96]: that looks better, lets go with this

**Review by AP-Hunt — APPROVED** [review]

**Pull request merged**